src/subscribers.ts

```diff
@@ -4,4 +4,12 @@ export let subscribers: ISubscriber[] = [];
 
 export function clearSubscribers() {
   subscribers = [];
+}
+
+export function removeSubscriberByAction(action: string) {
+  subscribers = subscribers.filter(subscriber => subscriber.action !== action);
+}
+
+export function removeSubscriberByCallback(callback: (args: any[]) => void) {
```

Comment:
We could probably handle the if statement now in the off() function of the mock, inside here. You could rename this to just 'removeSubscribers' and have the same signature as the off function` (action: string, callback?: (args: any[]) => void)`. That way you could handle both cases inside this module, having only one method for removal to expose.

Reply:
Still to look at this one.

Reply:
No problem. Let me know if I can help.

Reply:
Something I noticed on this one: [the callback function is not optional in the docs](https://docs.microsoft.com/en-us/javascript/api/@microsoft/signalr/hubconnection?view=signalr-js-latest#off-string---args--any-------void-), where in `SignalrMock`, it is.

I've made the callback non-optional as it is in the docs.

Reply:
Ah, good catch!